src/towncrier/_builder.py

```diff
@@ -246,6 +263,10 @@ def split_fragments(
                 # it's recorded.
                 content = ""
 
+            definitions[category]["underline_size"] = get_underline_size(
```

Comment:
The big problem with this code, is that the underline size should not be computed inside the `split_fragments` function

also, maybe call it `underline_lenght` ? but this is a very minor comment